docs/Contributing-back.rst

```diff
@@ -0,0 +1,610 @@
+
+.. _contributing_back_creating_pull_requests:
+
+Opening a new pull request
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. note:: This section describes the manual procedure to open a new pull request.
+
+          Please consider using ``eb --new-pr`` instead, see :ref:`github_new_pr`.
+
+.. note:: We are assuming you are already located in your local working copy of
+          the repository you want to contribute to (e.g., ``easybuild-framework``).
+
+To open a pull request for your contribution, you must follow these steps:
+
+i. :ref:`contributing_back_creating_pull_requests_branch`
+ii. :ref:`contributing_back_creating_pull_requests_commit`
+iii. :ref:`contributing_back_creating_pull_requests_push`
+iv. :ref:`contributing_back_creating_pull_requests_open_pr`
+
+
+.. _contributing_back_creating_pull_requests_branch:
+
+Creating a new branch
++++++++++++++++++++++
+
+First, create a new branch for your work. You can do this either before
+or after making the changes that you would like to contribute back, but we
+recommend to create a new branch before making any changes.
+
+Make sure you have the ``develop`` branch checked out before creating your branch:
+
+.. code:: shell
+
+  git checkout develop
+
+To create a new branch, you should use ``git branch <branch_name>`` followed by
+``git checkout <branch_name>``, or equivalently ``git checkout -b <branch_name>``.
+
+For example, to create a new branch named ``mybranch``:
+
+.. code:: shell
+
+  git checkout -b mybranch
+
+You can choose the branch name freely, but make it sufficiently descriptive --
+your future self will thank you.
+
+
+.. _contributing_back_creating_pull_requests_commit:
+
+Committing your changes
++++++++++++++++++++++++
+
+To 'save' your changes, you should create one or more *commits* in the branch
+you created. It is recommended to make separate commits for different 'units of work'.
+
+First, make sure you have checked out the branch were you want to commit the changes to.
+
+For example, to commit changes to the ``mybranch`` branch:
+
+.. code:: shell
+
+  git checkout mybranch
+
+To check which branch is currently checked out, use ``git branch``.
+
+To get a high-level overview of the changes before committing them, you can use ``git status``.
+
+To see the actual changes that were made, use ``git diff``.
+
+To commit the changes you want to contribute back, use ``git add <files>``
+to *stage* the changes, followed by ``git commit -m "<message>"`` to create
+the actual commit.
+
+For example, to commit the changes that were made to ``easybuild/tools/filetools.py``:
+
+.. code:: shell
+
+  git add easybuild/tools/filetools.py
+  git status  # check which files are staged for commit
+  git commit -m "example commit message for changes to filetools.py"
+
+.. note:: Please use a concise commit message that describes the changes you made.
+
+To create a single commit for *all* the changes that were made, use ``git commit -am "<message"``.
```

Comment:
ok, I overlooked the fact that files need to be tracked, I'll just drop this then